probe-rs/src/architecture/arm/sequences/atsame5x.rs

```diff
@@ -182,12 +185,23 @@ impl AtSAME5x {
 
     /// Perform a Chip-Erase operation
     ///
+    /// Issue a Chip-Erase command to the device provided that `permission` grants `erase-all`.
+    /// After successful erase the device will be reset into the appropriate state according to
+    /// `attach_method` i.e:
+    ///   * `AttachMethod::Normal` -> Normal HW Reset
+    ///   * `AttachMethod::UnderReset` -> Trigger CPU Reset Extension
+    ///
     /// # Errors
     /// This operation can fail due to insufficient permissions, or if Chip-Erase Lock is
     /// enabled (this lock can only be released from within the device firmare).
     /// After a successful Chip-Erase a `DebugProbeError::ReAttachRequired` is returned
     /// to signal that a re-connect is needed for the DSU to start operating in unlocked mode.
-    pub fn erase_all(&self, memory: &mut Memory, permissions: &Permissions) -> Result<(), Error> {
+    pub fn erase_all(
+        &self,
+        memory: &mut Memory,
+        permissions: &Permissions,
+        attach_method: AttachMethod,
```

Comment:
I will remove the attach_method argument, and instead check if it is in reset extension when at start and also reset back to reset extension if that is the case, otherwise do a normal reset at the end.